src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/CompoundTargetFrameworkValueProvider.cs

```diff
@@ -208,9 +212,17 @@ private async Task<string> ComputeValueAsync(ComplexTargetFramework complexTarge
             return targetFrameworkAlias;
         }
 
-        private static bool IsNetCore5OrHigher(string targetFrameworkAlias)
+        private static bool IsNetCore5OrHigher(string targetFrameworkAlias, string? targetFrameworkIdentifier)
         {
-            return targetFrameworkAlias.Contains("5.0") || targetFrameworkAlias.Contains("6.0") || targetFrameworkAlias.Contains("7.0");
+            // Converts the target framework alias to a double and checks if it's greater than or equal to 5.0.
+            // First, remove any non-numeric characters. Example: net5.0 => 5.0
+            targetFrameworkAlias = Regex.Replace(targetFrameworkAlias, "[^0-9.]", string.Empty);
+            double.TryParse(targetFrameworkAlias, out double targetFrameworkVersion);
+
+            if (targetFrameworkIdentifier is not null)
+                return targetFrameworkIdentifier.Contains(".NETCoreApp") && targetFrameworkVersion >= 5.0;
+
+            return false;
```

Comment:
I prefer this style to make it easier to read (at least for me).

Reply:
I *think* what Smit is suggesting is:
```suggestion
            return (targetFrameworkIdentifier?.Contains(".NETCoreApp") ?? false) && targetFrameworkVersion >= 5.0;
```

Which, I would agree, is a bit more clumsy to read. It does function the same way, though. You could do a ternary too, but you get almost the exact same readability.

*Note*: I do write statements like that (using `??` in boolean logic) on my own code because I like to keep things succinct. 😅

Reply:
That makes sense. I'll push some changes I've made since then; I think this will change a bit. Let's see if the readability there works better as an "extensive" block or "collapsed" into one line 🙂